test/fakeintake/aggregator/common.go

```diff
@@ -43,42 +46,40 @@ func newAggregator[P PayloadItem](parse parseFunc[P]) Aggregator[P] {
 	return Aggregator[P]{
 		payloadsByName: map[string][]P{},
 		parse:          parse,
+		mutex:          sync.RWMutex{},
 	}
 }
 
 // UnmarshallPayloads aggregate the payloads
 func (agg *Aggregator[P]) UnmarshallPayloads(payloads []api.Payload) error {
-	// reset map
-	agg.Reset()
-	// build map
+	// build new map
+	payloadsByName := map[string][]P{}
 	for _, p := range payloads {
 		payloads, err := agg.parse(p)
 		if err != nil {
 			return err
 		}
+
 		for _, item := range payloads {
-			if _, found := agg.payloadsByName[item.name()]; !found {
-				agg.payloadsByName[item.name()] = []P{}
+			if _, found := payloadsByName[item.name()]; !found {
+				payloadsByName[item.name()] = []P{}
 			}
-			agg.payloadsByName[item.name()] = append(agg.payloadsByName[item.name()], item)
+			payloadsByName[item.name()] = append(payloadsByName[item.name()], item)
 		}
 	}
+	agg.replace(payloadsByName)
 
 	return nil
 }
 
 // ContainsPayloadName return true if name match one of the payloads
 func (agg *Aggregator[P]) ContainsPayloadName(name string) bool {
-	_, found := agg.payloadsByName[name]
-	return found
+	return len(agg.GetPayloadsByName(name)) != 0
 }
 
 // ContainsPayloadNameAndTags return true if the payload name exist and on of the payloads contains all the tags
 func (agg *Aggregator[P]) ContainsPayloadNameAndTags(name string, tags []string) bool {
-	payloads, found := agg.payloadsByName[name]
-	if !found {
-		return false
-	}
+	payloads := agg.GetPayloadsByName(name)
```

Comment:
here you are changing the behaviour (you return `agg.payloadsByName[name]` whether it's found or not). I guess it returns `nil` when it's not found and so the next loop will parse nothing and the behaviour is fine, I just wanted to double check it's expected

Reply:
This comes from a previous comment. The loop will skip because `payloads` is nil and so it will still return false

Reply:
I will add a test with an empty aggregator

Reply:
Added !